common/src/main/java/com/microsoft/identity/common/internal/controllers/BaseNativeAuthController.kt

```diff
@@ -0,0 +1,250 @@
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.internal.controllers
+
+import com.microsoft.identity.common.java.cache.ICacheRecord
+import com.microsoft.identity.common.java.commands.parameters.CommandParameters
+import com.microsoft.identity.common.java.commands.parameters.DeviceCodeFlowCommandParameters
+import com.microsoft.identity.common.java.commands.parameters.GenerateShrCommandParameters
+import com.microsoft.identity.common.java.commands.parameters.InteractiveTokenCommandParameters
+import com.microsoft.identity.common.java.commands.parameters.RemoveAccountCommandParameters
+import com.microsoft.identity.common.java.commands.parameters.RopcTokenCommandParameters
+import com.microsoft.identity.common.java.commands.parameters.SilentTokenCommandParameters
+import com.microsoft.identity.common.java.controllers.BaseController
+import com.microsoft.identity.common.java.dto.AccountRecord
+import com.microsoft.identity.common.java.exception.ClientException
+import com.microsoft.identity.common.java.providers.microsoft.MicrosoftTokenResponse
+import com.microsoft.identity.common.java.providers.oauth2.AuthorizationRequest
+import com.microsoft.identity.common.java.providers.oauth2.AuthorizationResponse
+import com.microsoft.identity.common.java.providers.oauth2.AuthorizationResult
+import com.microsoft.identity.common.java.providers.oauth2.IAuthorizationStrategy
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2Strategy
+import com.microsoft.identity.common.java.providers.oauth2.OAuth2TokenCache
+import com.microsoft.identity.common.java.providers.oauth2.TokenResult
+import com.microsoft.identity.common.java.request.SdkType
+import com.microsoft.identity.common.java.result.AcquireTokenResult
+import com.microsoft.identity.common.java.result.GenerateShrResult
+import com.microsoft.identity.common.java.util.ported.PropertyBag
+import lombok.EqualsAndHashCode
+
+/**
+ * The implementation of the basis for MSAL native authentication.
+ * This class is introduced to conform with the method definition of BaseController, while none
+ * of these methods are relevant for native authentication flows. Hence, calling this logic in
+ * NativeAuthController will throw an exception.
+ *
+ * TODO: update BaseController and create a new layer in the hierarchy that contains non-native-auth
+ * logic. Doing this will make [BaseNativeAuthCommand] obsolete as well.
+ */
+@EqualsAndHashCode(callSuper = true, onlyExplicitlyIncluded = true)
+
+abstract class BaseNativeAuthController : BaseController() {
+
+    @Throws(Exception::class)
```

Comment:
I think it only throws `ClientException::class` as opposed to any generic Exception

Reply:
Same feedback for all the other places similar code is written